### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by cokeboL**
for this issue:
https://github.com/TarsCloud/TarsGo/issues/70

### CODE REVIEW

**Comment by pencil180430** [comment]
@maplebeats 
Do u mean timeout without close the chan?
It's ok not close a chan, test like this:

```golang
//test.go
package main

import (
	"time"
)

func testchan(ch chan int) {
	select {
	case ch <- 1:
	default:
	}
}

func main() {
	for {
		for i := 0; i < 1000000; i++ {
			ch := make(chan int)
			testchan(ch)
		}
		time.Sleep(time.Second / 20)
	}
}
```

```sh
GODEBUG=gctrace=1  go run test.go
```

**Comment by cokeboL** [comment]
@maplebeats 
Do u mean timeout without close the chan?
It's ok not close a chan, test like this:

```golang
//test.go
package main

import (
	"time"
)

func testchan(ch chan int) {
	select {
	case ch <- 1:
	default:
	}
}

func main() {
	for {
		for i := 0; i < 1000000; i++ {
			ch := make(chan int)
			testchan(ch)
		}
		time.Sleep(time.Second / 20)
	}
}
```

```sh
GODEBUG=gctrace=1  go run test.go
```

**Comment by maplebeats** [comment]
> 
> 
> @maplebeats
> Do u mean timeout without close the chan?
> It's ok not close a chan, readCh should be released by gc after
> 
> ```go
> adp.resp.Delete(msg.Req.IRequestId)
> ```
> 
> “net/rpc” also use chan in a similar way, and never close it.
> 
> and we may test like this:
> 
> ```go
> //test.go
> package main
> 
> import (
> 	"time"
> )
> 
> func testchan(ch chan int) {
> 	select {
> 	case ch <- 1:
> 	default:
> 	}
> }
> 
> func main() {
> 	for {
> 		for i := 0; i < 1000000; i++ {
> 			ch := make(chan int)
> 			testchan(ch)
> 		}
> 		time.Sleep(time.Second / 20)
> 	}
> }
> ```
> 
> ```shell
> GODEBUG=gctrace=1  go run test.go
> ```

you are right, 3ks 👍

**Comment by defool** [comment]
LGTM

**Comment by cokeboL** [comment]
ok, I'll close #70

File: tars/adapter.go [hunk 1]
```diff
@@ -72,7 +65,11 @@ func (c *AdapterProxy) Recv(pkg []byte) {
 	if ok {
 		ch := chIF.(chan *requestf.ResponsePacket)
 		TLOG.Debug("IN:", packet)
-		ch <- &packet
+		select {
+		case ch <- &packet:
```
Comment (maplebeats) [inline tars/adapter.go]: when the packet return before the msg subscibe，
the channel will be closed and readCh will return nil to msg。

Comment (cokeboL) [reply tars/adapter.go]: one more pr https://github.com/TarsCloud/TarsGo/pull/73

**Pull request merged**